postreise/plot/plot_bar_generation_stack.py

```diff
@@ -0,0 +1,234 @@
+import os
+
+import matplotlib.pyplot as plt
+import pandas as pd
+from powersimdata.network.model import ModelImmutables, area_to_loadzone
+from powersimdata.scenario.scenario import Scenario
+
+from postreise.analyze.generation.curtailment import (
+    calculate_curtailment_time_series_by_areas_and_resources,
+)
+from postreise.analyze.generation.summarize import sum_generation_by_type_zone
+
+
+def plot_bar_generation_stack(
+    areas,
+    scenario_ids,
+    resources,
+    area_types=None,
+    scenario_names=None,
+    curtailment_split=True,
+    t2c=None,
+    t2l=None,
+    t2hc=None,
+    titles=None,
+    plot_show=True,
+    save=False,
+    filenames=None,
+    filepath=None,
+):
+    """Plot any number of scenarios as generation stack bar for selected resources in
+    each specified areas.
+
+    :param list/str areas: list of area(s), each area is one of *loadzone*, *state*,
+        *state abbreviation*, *interconnect*, *'all'*.
+    :param int/list/str scenario_ids: list of scenario id(s), defaults to None.
+    :param str/list resources: one or a list of resources. *'curtailment'*,
+        *'solar_curtailment'*, *'wind_curtailment'*, *'wind_offshore_curtailment'*
+        are valid entries together with all available generator types in the area(s).
+        The order of the resources determines the stack order in the figure.
+    :param list/str area_types: list of area_type(s), each area_type is one of
+        *'loadzone'*, *'state'*, *'state_abbr'*, *'interconnect'*, defaults to None.
+    :param list/str scenario_names: list of scenario name(s) of same len as scenario
+        ids, defaults to None.
+    :param bool curtailment_split: if curtailments are split into different
+        categories, defaults to True.
+    :param dict t2c: user specified color of resource type to overwrite type2color
```

Comment:
I think for the user who might not know `type2color`, I would just say: `overwrite pre-defined colors`

Reply:
Done.